src/dropdown/dropdown.ts

```diff
@@ -242,7 +252,63 @@ export class NgbDropdown implements OnInit, OnDestroy {
 
   private _positionMenu() {
     if (this.isOpen() && this._menu) {
-      this._menu.position(this._anchor.anchorEl, this.placement);
+      this._position(this._anchor.anchorEl, this.placement, this.container);
+    }
+  }
+
+  private _position(triggerEl, placement, container) {
+    this._applyPlacement(
+        positionElements(
+            triggerEl, this._bodyContainer || this._menuElement.nativeElement, placement, container === 'body'));
+  }
+
+  private _applyContainer(container: null | 'body' = null) {
```

Comment:
I would split this method at least in two:
- one that does the setup
- another that does the cleanup

Currently cleanup is confusing (you have to pass `container = null` to do it?)